source/extensionPoints/__init__.py

```diff
@@ -66,22 +70,28 @@ def notifyOnce(self, **kwargs):
 				log.exception(f"Error running handler {handler} for {self}. Exception {e}")
 
 
-FilterValueTypeT = TypeVar("FilterOutputTypeT")
+FilterValueTypeT = TypeVar("FilterValueTypeT")
 
 
-class Filter(HandlerRegistrar, Generic[FilterValueTypeT]):
+# Typing doesn't allow us to typehint a Callable with one required argument and several kwargs.
+# Using a typing_extensions.Protocol for this results in false negatives.

```

Comment:
Have you considered `typing_extensions.ParamSpec`? I'm not sure if it is appropriate here, but it may be more applicable than `Protocol`.

Reply:
A great, I haven't thought about typing_extensions. Definitely going to check that out.

Reply:
Turns out it is not appropriate. It requires both *args and **kwargs be present on a callable, and that's not the case in case of our extension points.